lib/jobs/switch-node-relations.js

```diff
@@ -0,0 +1,212 @@
+     */
+    SwitchRelationsJob.prototype._updateSwitchRelations =
+        function updateSwitchRelations(lldpData, info) {
+            var self = this;
+            return waterline.nodes.findByIdentifier(lldpData.node)
+            .then(function (node) {
+                if (!node) {
+                    return Promise.reject('Could not find node with identifier ' + lldpData.node);
+                }
+                else {
+                    return Promise.resolve(node.relations);
+                }
+            })
+            .filter(function (entry) {
+                return (entry.targets.indexOf(self.nodeId) < 0);
+            })
+            .then(function (relations) {
+                relations.push({
+                    relationType: RELATION_TYPE,
+                    targets: [self.nodeId],
+                    info: info
+                });
+                return waterline.nodes.updateByIdentifier(
+                    lldpData.node,
+                    { relations: relations });
+            });
+
+        };
+
+    function _getCatalogsBySource(src) {
+        return waterline.catalogs.find({
+            source: src
+        });
+    }
+
+    /**
+     * Iterate through switch mac addresses list
+     * for a match in nodes lldp catalog
+     * return a promise
+     */
+    function _findMatchingMac(nodeLldpData, lldpMacList) {
+        var relationsList = {};
+        return Promise.all(
+        _.forEach(_.keys(nodeLldpData.data), function (ethPort) {
+            //validate mac first
+            var lldpMac = catalogSearch.getPath(nodeLldpData.data[ethPort], 'chassis.mac');
+            if (!lldpMac) {
+                throw new Error('cant find mac address in lldp data for node ' +
+                    nodeLldpData.node + ' port ' + ethPort);
+            }
+            if (_validateMacStrFormat(lldpMac)) {
+                lldpMac = _adjustMacStrFormat(lldpMac);
+                if (lldpMacList.indexOf(lldpMac) >= 0) {
+                    var singleEntry = {
+                        destMac: nodeLldpData.data[ethPort].chassis.mac,
+                        destPort: nodeLldpData.data[ethPort].port.ifname
+                    };
+                    relationsList[ethPort] = singleEntry;
+                }
+            } else {
+                throw new Error('mac: ' + lldpMac + ' bad format',
+                    +nodeLldpData.node + ' port ' + ethPort);
+            }
+        }))
+        .then(function () {
+            return Promise.resolve(relationsList);
+        });
+    }
+
+    /**
+     * verify that mac address is 5 hex nbs separated by ':'
+     * returns a booleen
+     */
+    function _validateMacStrFormat(macString) {
+        var regex = /^(([A-Fa-f0-9]{1,2}[:]){5}[A-Fa-f0-9]{1,2}[,]?)+$/;
+        return regex.test(macString);
+    }
+
+    /**
+     * Adjust mac Address format to 2 hex digits
+     * return mac address string
+     */
+    function _adjustMacStrFormat(macString) {
```

Comment:
@keedya Do you also need to normalize case to compare macs?  Your regex accepts both upper and lower case hex digits.